oak_runtime/src/node/http/client.rs

```diff
@@ -0,0 +1,327 @@
+    /// to handle URIs with the `HTTP` scheme, then it should be labelled public. In this case,
+    /// the authority could be empty, meaning that the node can handle all HTTP requests. In
+    /// addition, a non-empty `root_ca` is always required.
+    pub fn new(node_name: &str, config: HttpClientConfiguration, root_ca: Vec<u8>) -> Option<Self> {
+        if !validate_config(&config, root_ca.is_empty()) {
+            return None;
+        }
+        let http_client = create_client(root_ca).ok()?;
+        let node_privilege = get_privilege(config.authority.clone());
+        let authority = Some(config.authority).filter(|s| !s.is_empty());
+        Some(HttpClientNode {
+            node_name: node_name.to_string(),
+            http_client,
+            authority,
+            node_privilege,
+        })
+    }
+
+    /// Main loop that handles HTTP invocations from the `handle`, sends HTTP requests to an
+    /// external HTTP service and writes HTTP responses back to the invocation channel.
+    async fn handle_loop(&mut self, runtime: RuntimeProxy, handle: Handle) -> Result<(), OakError> {
+        // Create a [`Receiver`] to receive the HTTP request.
+        let receiver = Receiver::<Invocation>::new(ReadHandle { handle });
+        loop {
+            debug!("Waiting for HTTP invocation");
+            // Read an HTTP invocation from the [`Receiver`].
+            let invocation = receiver.receive(&runtime).map_err(|error| {
+                match error {
+                    OakError::OakStatus(OakStatus::ErrTerminated) => {
+                        debug!("HTTP client node is terminating.")
+                    }
+                    OakError::OakStatus(OakStatus::ErrChannelClosed) => {
+                        info!("HTTP invocation channel closed")
+                    }
+                    _ => error!("Couldn't receive the invocation: {:?}", error),
+                }
+                error
+            })?;
+
+            self.process_invocation(&runtime, &invocation).await;
+            debug!("Invocation processing finished");
+            invocation.close(&runtime);
+        }
+    }
+
+    /// Process an HTTP invocation for an external HTTP service.
+    async fn process_invocation(&mut self, runtime: &RuntimeProxy, invocation: &Invocation) {
+        // TODO(#1717): Update HTTP client metrics
+
+        // A closure to ease sending and logging the errors
+        let send_error = |err, code, msg| {
+            warn!("{}: {}", msg, err);
+            invocation.send_error(code, msg, runtime);
+        };
+
+        // Receive a request from the invocation channel.
+        let request = match invocation.receive_request(&runtime) {
+            Ok(request) => request,
+            Err(err) => {
+                send_error(
+                    format!("{:?}", err),
+                    http::StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
Not sure I understand where these errors end up, does it mean that a caller would get these errors as if they are coming from the remote server, even though they are actually coming from this pseudo-node implementation? (I don't think that needs to change in this PR anyways, but if that's the case we may want to think how to better propagate real HTTP errors from internal node errors at some point perhaps).